File: ErrorBot.php
<?php
// ErrorBot integration for PHP

class ErrorBot {
    private $apiKey;
    private $projectName;
    private $endpoint = 'https://errorbot.fyi/api/v1/report';

    public function __construct($apiKey, $projectName) {
        $this->apiKey = $apiKey;
        $this->projectName = $projectName;
    }

    public function init() {
        set_exception_handler([$this, 'handleException']);
        set_error_handler([$this, 'handleError']);
    }

    public function handleException($exception) {
        $this->reportError($exception->getMessage(), 'error');
    }

    public function handleError($errno, $errstr, $errfile, $errline) {
        $this->reportError("$errstr in $errfile on line $errline", 'error');
    }

    public function reportError($message, $type = 'error') {
        $errorData = [
            'message' => $message,
            'type' => $type,
            'project' => $this->projectName
        ];

        $ch = curl_init($this->endpoint);
        curl_setopt($ch, CURLOPT_POST, 1);
        curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($errorData));
        curl_setopt($ch, CURLOPT_HTTPHEADER, [
            'Content-Type: application/json',
            "X-API-Key: {$this->apiKey}"
        ]);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

        $response = curl_exec($ch);
        if (curl_errno($ch)) {
            error_log('Failed to report error to ErrorBot: ' . curl_error($ch));
        } else {
            $result = json_decode($response, true);
            if ($result['success']) {
                error_log('Error reported successfully');
            } else {
                error_log('Failed to report error: ' . $result['error']['message']);
            }
        }
        curl_close($ch);
    }
}

// Usage
$errorBot = new ErrorBot('YOUR_API_KEY_HERE', 'Your Project Name');
$errorBot->init();

// To manually report an error
// $errorBot->reportError('Something went wrong', 'warning');

File: config/initializers/error_bot.rb
# ErrorBot integration for Ruby on Rails

require 'net/http'
require 'json'

class ErrorBot
  def self.setup
    api_key = 'YOUR_API_KEY_HERE'
    project_name = 'Your Project Name'
    endpoint = 'https://errorbot.fyi/api/v1/report'

    Rails.application.config.middleware.use ExceptionNotification::Rack,
      :error_bot => {
        :api_key => api_key,
        :project => project_name,
        :endpoint => endpoint
      }
  end

  def self.notify(exception, env = nil)
    error_data = {
      message: exception.message,
      type: 'error',
      project: Rails.application.config.exception_notification[:error_bot][:project]
    }

    uri = URI(Rails.application.config.exception_notification[:error_bot][:endpoint])
    http = Net::HTTP.new(uri.host, uri.port)
    http.use_ssl = (uri.scheme == 'https')

    request = Net::HTTP::Post.new(uri.path, {
      'Content-Type' => 'application/json',
      'X-API-Key' => Rails.application.config.exception_notification[:error_bot][:api_key]
    })
    request.body = error_data.to_json

    response = http.request(request)
    
    if response.is_a?(Net::HTTPSuccess)
      result = JSON.parse(response.body)
      if result['success']
        Rails.logger.info "Error reported successfully to ErrorBot"
      else
        Rails.logger.error "Failed to report error to ErrorBot: #{result['error']['message']}"
      end
    else
      Rails.logger.error "Failed to report error to ErrorBot: HTTP #{response.code}"
    end
  end
end

ErrorBot.setup